### DISCUSSION TITLE
Rfid Esp32 gateway (olimex)

### DISCUSSION THREAD

**Discussion by Farnibull** [OpenEVSE/openevse_esp32_firmware#802] [General]
Hello, I have been using the OpenEVSE terminal for a while with RFID. It works well. I ordered the ESP32 gateway board for the Ethernet connection. Where should I connect my PN532 RFID to my ESP32 gateway on the ESP32 gateway? (SCL/SDA?)
I imagine that in the Gpio PIN code is defined in esp32. when you load version 5.0.0 on an ESP32-GATEWAY are the PINs the same? how does it happen?
Thanks for your help

**Comment by jeremypoulter** [comment]
There are a few build options you need to add to enable RFID. You can look at the `openevse_wifi_v1` env in `platformio.ini` for an example, but we haven't formally enabled RFID for the ESP32 Gateway board. The main reason for this is that although the board has a large amount of GPIO pins easily accessible they are basically useless as they are input only or used by the internal flash chip IIRC. That being said I haven't reviewed this in a while and the latest revision F board may be different.

**Reply by Farnibull** [reply]
thank you for your answer, so unfortunately I don't really know the programming systems on this card very well. But I will try.
I therefore understand that version 5.0.0 ESP32-gateway and compatibility with ESP32-gateway (I received with this version) but the RFID systems are not implemented?
one day do you think that RFID will be on the esp32-gateway?
thank you for your valuable help.

**Reply by jeremypoulter** [reply]
It is not something we actively have been looking at, but would be happy to accept a contribution. As I mentioned the main issue is that at least some of those boards don't have enough pins exposed on the GPIO connector to enable you to connect the RFID reader. The first step is to review this paying particular attention to which pins are used by other hardware and if they are input only.
